File: src/types/draft-wikinode-suggestion.ts
import type { ParsedDocumentSourceRef } from "@/types/raw-material"
import type { KnowledgeObjectType } from "@/types/wiki"

export type DraftWikiNodeRelationCandidate = {
  targetTitle: string
  relationType: string
  source: string
  confidence?: number
}

export type DraftWikiNodeSuggestion = {
  suggestionId: string
  parsedDocumentId: string
  rawMaterialId: string
  sourceId: string
  operationId: string
  title: string
  objectType: KnowledgeObjectType
  subtype?: string
  contentDraft: string
  metadataDraft: Record<string, string>
  sourceRefs: ParsedDocumentSourceRef[]
  relationCandidates: DraftWikiNodeRelationCandidate[]
  confidence?: number
  status: "draft" | "needs_review" | "accepted" | "rejected" | "superseded"
  reviewNote?: string | null
  conflictStatus: "none" | "title_match" | "source_ref_match" | "existing_suggestion" | "accepted_before"
  conflictReasons: string[]
  matchedWikiNodeIds: string[]
  matchedSuggestionIds: string[]
  sourceRefCount: number
  relationCandidateCount: number
  createdAt: string
  updatedAt: string
}

File: src/main/java/com/wikinode/studio/model/DraftWikiNodeSuggestion.java
package com.wikinode.studio.model;

import java.util.List;
import java.util.Map;

public record DraftWikiNodeSuggestion(
  String suggestionId,
  String parsedDocumentId,
  String rawMaterialId,
  String sourceId,
  String knowledgeBaseId,
  String operationId,
  String title,
  String objectType,
  String subtype,
  String contentDraft,
  Map<String, String> metadataDraft,
  List<ParsedDocumentSourceRef> sourceRefs,
  List<DraftWikiNodeRelationCandidate> relationCandidates,
  Double confidence,
  String status,
  String reviewNote,
  String conflictStatus,
  List<String> conflictReasons,
  List<String> matchedWikiNodeIds,
  List<String> matchedSuggestionIds,
  int sourceRefCount,
  int relationCandidateCount,
  String createdAt,
  String updatedAt
) {
  public DraftWikiNodeSuggestion(
    String suggestionId,
    String parsedDocumentId,
    String rawMaterialId,
    String sourceId,
    String operationId,
    String title,
    String objectType,
    String subtype,
    String contentDraft,
    Map<String, String> metadataDraft,
    List<ParsedDocumentSourceRef> sourceRefs,
    List<DraftWikiNodeRelationCandidate> relationCandidates,
    Double confidence,
    String status,
    String reviewNote,
    String conflictStatus,
    List<String> conflictReasons,
    List<String> matchedWikiNodeIds,
    List<String> matchedSuggestionIds,
    int sourceRefCount,
    int relationCandidateCount,
    String createdAt,
    String updatedAt
  ) {
    this(
      suggestionId,
      parsedDocumentId,
      rawMaterialId,
      sourceId,
      null,
      operationId,
      title,
      objectType,
      subtype,
      contentDraft,
      metadataDraft,
      sourceRefs,
      relationCandidates,
      confidence,
      status,
      reviewNote,
      conflictStatus,
      conflictReasons,
      matchedWikiNodeIds,
      matchedSuggestionIds,
      sourceRefCount,
      relationCandidateCount,
      createdAt,
      updatedAt
    );
  }
}
